File: ColegioTrenerAndroid/app/src/main/java/com/jjmf/colegiotrenerandroid/data/dto/PagoDto.kt
package com.jjmf.colegiotrenerandroid.data.dto

import com.google.gson.annotations.SerializedName
import com.jjmf.colegiotrenerandroid.domain.model.Pago
import com.jjmf.colegiotrenerandroid.util.toDate
import java.util.Date

data class PagoDto(
    @SerializedName("anoaca") val anoaca: String?,
    @SerializedName("codconcepto") val codconcepto: String?,
    @SerializedName("concepto") val concepto: String?,
    @SerializedName("fecven") val fecven: String?,
    @SerializedName("saldo") val saldo: Double?,
    @SerializedName("estado") val estado: String?,
    @SerializedName("tipdoc") val tipdoc: String?,
    @SerializedName("numdoc") val numdoc: String?,
    @SerializedName("mediopago") val mediopago: String?,
    @SerializedName("fecpag") val fecpag: String?,
    @SerializedName("importepagado") val importepagado: Double?,
    @SerializedName("mora") val mora: Double?,
) {
    fun toDomain(): Pago {
        return Pago(
            anoaca = anoaca?.trim(),
            codconcepto = codconcepto?.trim(),
            concepto = concepto?.trim(),
            fecven = fecven?.trim()?.toDate() ?: Date(),
            saldo = saldo ?: 0.0,
            estado = estado?.trim(),
            tipdoc = tipdoc?.trim(),
            numdoc = numdoc?.trim(),
            mediopago = mediopago?.trim(),
            fecpag = fecpag?.trim()?.toDate() ?: Date(),
            importepagado = importepagado ?: 0.0,
            mora = mora ?: 0.0
        )
    }
}

File: ColegioTrenerSwift/ColegioTrenerSwift/Network/Dto/PagoDto.swift
//
//  PagoDto.swift
//  ColegioTrenerSwift
//
//  Created by Joel Maldonado on 25/04/24.
//

import SwiftUI


struct PagoDto: Codable {
    let anoaca: String?
    let codconcepto: String?
    let concepto: String?
    let fecven: String?
    let saldo: Double?
    let estado: String?
    
    //Realizado
    let tipdoc: String?
    let numdoc: String?
    let mediopago: String?
    let fecpag: String?
    let importepagado: Double?
    let mora: Double?
    
    func toDomain() -> Pago {
        return Pago(
            anoaca: anoaca?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            codconcepto: codconcepto?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            concepto: concepto?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            fecven: fecven?.toDate() ?? .now,
            saldo: saldo ?? 0.0,
            estado: estado?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            tipdoc: tipdoc?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            numdoc: numdoc?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            mediopago: mediopago?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            fecpag: fecpag?.toDate() ?? .now,
            importepagado: importepagado ?? 0.0,
            mora: mora ?? 0.0
        )
    }
}
